platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCase.java

```diff
@@ -0,0 +1,219 @@
+        assertThat(email.getBody(), containsString(BODY));
+        assertThat(email.getBody(), containsString(ATTACHMENT_TEXT));
+        assertThat(email.getBody(), containsString(ATTACHMENT_FILENAME));
+
+    }
+
+    @Test
+    public void testSend() throws IOException, MessagingException {
+
+        int port = findAvailablePort();
+
+        SimpleSmtpServer server = SimpleSmtpServer.start(port);
+
+        SmtpClientImpl emailService = new SmtpClientImpl();
+
+        emailService.setHostName(HOSTNAME);
+        emailService.setPortNumber(port);
+
+        Session session = emailService.createSession();
+
+        MimeMessage mimeMessage = new MimeMessage(session);
+        mimeMessage.setFrom(new InternetAddress(FROM_ADDR));
+        mimeMessage.addRecipient(Message.RecipientType.TO, new InternetAddress(TO_ADDR));
+        mimeMessage.setSubject(SUBJECT);
+
+        BodyPart messageBodyPart = new MimeBodyPart();
+        messageBodyPart.setText(BODY);
+
+        Multipart multipart = new MimeMultipart();
+        multipart.addBodyPart(messageBodyPart);
+
+        mimeMessage.setContent(multipart);
+
+        emailService.send(mimeMessage);
+
+        server.stop();
+
+        assertThat(server.getReceivedEmailSize(), is(1));
+        Iterator emailIterator = server.getReceivedEmail();
+        SmtpMessage email = (SmtpMessage) emailIterator.next();
+        assertThat(email.getHeaderValue(SUBJECT_HEADER), is(SUBJECT));
+        assertThat(email.getHeaderValue(FROM_HEADER), containsString(FROM_ADDR));
+        assertThat(email.getHeaderValue(TO_HEADER), containsString(TO_ADDR));
+        assertThat(email.getBody(), containsString(BODY));
+
+    }
+
+    /**
+     * There isn't a great way to test that authenticated email sending works in
+     * a unit test, but we can at least make sure the PasswordAuthentication object
+     * in the Session has the correct username and password.
+     */
+    @Test
+    public void testWithUsernamePassword() throws UnknownHostException {
+
+        String username = "username";
+        String password = "password";
+
+        SmtpClientImpl emailService = new SmtpClientImpl();
+
+        emailService.setHostName("host.com");
+        emailService.setPortNumber(25);
+        emailService.setUserName(username);
+        emailService.setPassword(password);
+
+        Session session = emailService.createSession();
+
+        PasswordAuthentication actual = session.requestPasswordAuthentication(InetAddress.getByName(
+                "127.0.0.1"), 25, "smtp", "prompt", "defaultUserName");
+
+        assertThat(actual.getUserName(), is(username));
+        assertThat(actual.getPassword(), is(password));
+
+    }
+
+    private int findAvailablePort() throws IOException {
+        try (ServerSocket s = new ServerSocket(0)) {
+            return s.getLocalPort();
+        }
+    }
+
+    private static class DeletableFile implements Closeable {
```

Comment:
❗️ We should be able to use http://junit.org/junit4/javadoc/4.12/index.html?org/junit/rules/TemporaryFolder.html instead of creating our own `DeletableFile`.